File: langgraph/data/semantic_equiv/81.py
def convert_gpa_to_letter(scores):

    grade_letters = []
    index = 0
    while index < len(scores):
        gpa = scores[index]
        if gpa == 4.0:
            grade_letters.append("A+")
        elif gpa > 3.7:
            grade_letters.append("A")
        elif gpa > 3.3:
            grade_letters.append("A-")
        elif gpa > 3.0:
            grade_letters.append("B+")
        elif gpa > 2.7:
            grade_letters.append("B")
        elif gpa > 2.3:
            grade_letters.append("B-")
        elif gpa > 2.0:
            grade_letters.append("C+")
        elif gpa > 1.7:
            grade_letters.append("C")
        elif gpa > 1.3:
            grade_letters.append("C-")
        elif gpa > 1.0:
            grade_letters.append("D+")
        elif gpa > 0.7:
            grade_letters.append("D")
        elif gpa > 0.0:
            grade_letters.append("D-")
        else:
            grade_letters.append("E")
        index += 1
    return grade_letters

# method name: numerical_letter_grade -> convert_gpa_to_letter
# parameter name: grades -> scores
# variable name: letter_grade -> grade_letters
# replaced for loop with while loop
# introduced index variable to iterate through scores list

File: langgraph/result/qwen/3b-py2js/81.js
function numericalLetterGrade(grades) {
    let letterGrade = [];
    for (let gpa of grades) {
        if (gpa === 4.0) {
            letterGrade.push("A+");
        } else if (gpa > 3.7) {
            letterGrade.push("A");
        } else if (gpa > 3.3) {
            letterGrade.push("A-");
        } else if (gpa > 3.0) {
            letterGrade.push("B+");
        } else if (gpa > 2.7) {
            letterGrade.push("B");
        } else if (gpa > 2.3) {
            letterGrade.push("B-");
        } else if (gpa > 2.0) {
            letterGrade.push("C+");
        } else if (gpa > 1.7) {
            letterGrade.push("C");
        } else if (gpa > 1.3) {
            letterGrade.push("C-");
        } else if (gpa > 1.0) {
            letterGrade.push("D+");
        } else if (gpa > 0.7) {
            letterGrade.push("D");
        } else if (gpa > 0.0) {
            letterGrade.push("D-");
        } else {
            letterGrade.push("E");
        }
    }
    return letterGrade;
}